scripts/drs_v1.1_uri_migration.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+import csv
+from firecloud import api as fapi
+from io import StringIO
+import json
+import re
+
+GUID_PATTERN = re.compile(r'^[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}$', re.IGNORECASE)
+
+
+def update_entities_to_compact_identifier(workspace, project, single_etypes_list, dry_run):
+    """Update Data Model entity attributes to DRS 1.1 Compact Identifier."""
+
+    for etype in single_etypes_list:
+        print(f'Starting TCGA DRS updates for entity: {etype}')
+
+        # get entity table response for API call
+        res_etype = fapi.get_entities_tsv(project, workspace, etype, model="flexible")
+
+        # save current/original data model tsv files for provenance
+        print(f'Saving original {etype} TSV.')
+        original_tsv_name = f"original_{etype}_{project}-{workspace}_table.tsv"
+        with open(original_tsv_name, "w") as f:
+            f.write(res_etype.text)
+
+        # read entity table response into dictionary to perform DRS URL updates
+        dict_etype = list(csv.DictReader(StringIO(res_etype.text), delimiter='\t'))
+
+        # create empty list to add updated rows and list to capture list of columns that were modified
+        drs_dict_table = []
+        modified_cols = set()
+        # for "row" (each row is [list] of column:values)
+        for row in dict_etype:
+            drs_row = row.copy()
+            # for each column in row
+            for col in row:
+                # check if the col values are dataguids.org URLs and parse out guid
+                if row[col].startswith("drs://dataguids.org"):
+                    guid = row[col].split("/")[3]
+                    # only modify col if guid is valid and exists
+                    if guid and GUID_PATTERN.match(guid):
+                        drs_url = "drs://dg.4DFC:" + guid
+                        drs_row[col] = drs_url
+                        modified_cols.add(col)
+
+            # append new "row" with updated drs values to new list
+            drs_dict_table.append(drs_row)
+
+        # save new/drs updated data model tsv files for provenance
+        print(f'Saving updated {etype} TSV.')
+        updated_tsv_name = f"updated_{etype}_{project}-{workspace}_table.tsv"
+        tsv_headers = drs_dict_table[0].keys()
+
+        with open(updated_tsv_name, 'w') as outfile:
+            # get keys from OrderedDictionary and write rows, separate with tabs
+            writer = csv.DictWriter(outfile, tsv_headers, delimiter="\t")
+            writer.writeheader()
+            writer.writerows(drs_dict_table)
+
+        # list of the columns that are scoped to be updated if re-run with dry_run = False
+        modified_cols = list(modified_cols)
+        if dry_run:
+            print(f"Columns in the {etype} table that *will be* be updated when notebook is re-run with `dry_run = False`:")
```

Comment:
here this should be `when this script is re-run without the --dry-run flag` right?

Reply:
ah yes the instructions in there are for the notebook. good catch.